Src/Plugins/Security/SecurePlaintextSecretsValidators/AwsHttpRequestSigner.cs

```diff
@@ -0,0 +1,142 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Security.Cryptography;
+using System.Text;
+
+using Microsoft.CodeAnalysis.Sarif.PatternMatcher.Sdk;
+
+namespace Microsoft.CodeAnalysis.Sarif.PatternMatcher.Plugins.Security
+{
+    // https://docs.aws.amazon.com/general/latest/gr/create-signed-request.html
+
+    internal class AwsHttpRequestSigner : IDisposable
+    {
+        private const string ISO8601BasicDateTimeFormat = "yyyyMMddTHHmmssZ";
+
+        private readonly SHA256 sha256;
+        private readonly string accessKey;
+        private readonly string accessSecret;
+
+        public AwsHttpRequestSigner(string accessKey, string accessSecret)
+        {
+            this.accessKey = accessKey;
+            this.accessSecret = accessSecret;
+            this.sha256 = SHA256.Create();
+        }
+
+        public virtual string Version => "AWS4";
+
+        public virtual string RequestType => "aws4_request";
+
+        public virtual string Algorithm => "AWS4-HMAC-SHA256";
+
+        public virtual string HostHeaderName => "host";
+
+        public virtual string DateHeaderName => "x-amz-date";
+
+        public virtual string ContentSha256HeaderName => "x-amz-content-sha256";
+
+        public void SignRequest(HttpRequestMessage request,
+                                string region,
+                                string service,
+                                DateTime? dateTime = null)
```

Comment:
>dateTime [](http://example.com/codeflow?start=42&length=8)

used in unit test mocking, the mock setup http request and actual http request setup need to have same timestamp.